### PR TITLE
N21-420 CTL Controller

### PR DESCRIPTION
**Pull request by MarvinOehlerkingCap**
# Description
- call nest api for ctl tools

## Links to Tickets or other pull requests
https://github.com/hpi-schul-cloud/superhero-dashboard/pull/185
https://ticketsystem.dbildungscloud.de/browse/N21-239
https://ticketsystem.dbildungscloud.de/browse/N21-420

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: controllers/tools.js [hunk 1]
```diff
@@ -64,6 +64,7 @@ const sanitizeTool = (req, create=false) => {
   req.body.openNewTab = !!req.body.openNewTab;
   req.body.frontchannel_logout_uri = req.body.frontchannel_logout_uri || null;
   req.body.isHidden = !!req.body.isHidden;
+  console.log(req.body);
```
Comment (arnegns) [inline controllers/tools.js:67]: You forgot to delete this :P

File: ansible/roles/superhero-dashboard/templates/configmap.yml.j2 [hunk 2]
```diff
@@ -10,3 +10,5 @@ data:
   SC_NAV_TITLE: "{{ SC_NAV_TITLE }}"
   HOST: "https://{{ DOMAIN }}"
   FEATURE_LEGACY_HYDRA_ENABLED: "{{ FEATURE_LEGACY_HYDRA_ENABLED|default(false, true) }}"
+  FEATURE_CTL_TOOLS_ENABLED: "{{ FEATURE_CTL_TOOLS_ENABLED|default(false, true) }}"
+  FEATURE_LTI_TOOLS_DISABLED: "{{ FEATURE_LTI_TOOLS_DISABLED|default(false, true) }}"
```
Comment (arnegns) [inline ansible/roles/superhero-dashboard/templates/configmap.yml.j2:14]: Maybe a topic for the arc group :P @alweber-cap

File: views/ctltools/ctltools.hbs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    {{#content "scripts" mode="append"}}
+        <script src="/scripts/ctltools.js" type="text/javascript" defer></script>
+    {{/content}}
+
+    {{#content "page"}}
+        <div class="container-fluid">
+            <div class="row mobile-no-margin-right">
+                <div class="card h-100">
+                    <div class="card-block">
+                        <h5 class="card-title">OAuth2-Endpunkte</h5>
```
Comment (arnegns) [inline views/ctltools/ctltools.hbs:16]: I think we should remove this informations

**Review by arnegns — DISMISSED** [review]

File: controllers/ctltools.js [hunk 4]
```diff
@@ -2,301 +2,225 @@
...
-        });
-      } else {
-        createTool(req, service, next);
-      }
-    };
+const getDetailHandler = (req, res, next) => {
+    api(req, { version: 'v3' }).get(`/tools/${req.params.id}`).then(data => {
+        if (data.config.type === 'oauth2') {
+            data.config.redirectUris = data.config.redirectUris.join(';');
+            data.config.clientSecret = PASSWORD;
```
Comment (Metauriel) [inline controllers/ctltools.js:84]: I assume this is the prefill the user sees in the password field? is there no better way to do this on the field itself? I cant believe its a best practise to manually set this on pageload, and then sanitize it on submit...

**Review by Metauriel — CHANGES_REQUESTED** [review]

File: controllers/ctltools.js (hunk 4 above)
Comment (alweber-cap) [reply controllers/ctltools.js:84]: it was the previous practice to do it like that. This is just copied code for now. Once we start to properly implement the new functionalities we will take a look into improving this code

**Review by Metauriel — DISMISSED** [review]

**Review by IgorCapCoder — APPROVED** [review]

**Pull request merged**